1.15.0 has no `py.typed`

This leads to complaints from mypy and basedpyright in scipy-stubs:

```
scipy-stubs/_lib/_array_api.pyi:11: error: Skipping analyzing "array_api_compat": module is installed, but missing library stubs or py.typed marker  [import-untyped]
scipy-stubs/_lib/_array_api.pyi:11: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 821 source files)
```

```
/home/joren/Workspace/scipy-stubs/scipy-stubs/_lib/_array_api.pyi
  /home/joren/Workspace/scipy-stubs/scipy-stubs/_lib/_array_api.pyi:11:6 - error: Stub file not found for "array_api_compat" (reportMissingTypeStubs)
1 error, 0 warnings, 0 notes
```

--- a/meson.build
+++ b/meson.build
@@ -11,6 +11,7 @@ sources_raw = {
   'array_api_compat': [
     'src/array_api_compat/__init__.py',
     'src/array_api_compat/_internal.py',
+    'src/array_api_compat/py.typed',
   ],
 
   'array_api_compat/common': [

